### ISSUE TITLE
Not actionable error about unaccurate Scala 2 existential type

### ISSUE THREAD

**Issue by cchantep** [lampepfl/dotty#14272]
## Compiler version

3.1.0

## Minimized code

```scala
// compile with 2.13.7
import language.existentials

abstract class Box[T]

object Box {
  def id(b: Box[Box[x]] forSome { type x }) = b
}
```

```scala
// compile with 3.1.0
@main def test =
  val b = Box.id(??? : Box[Box[Int]])
```

## Output

```
An existential type that came from a Scala-2 classfile cannot be
mapped accurately to to a Scala-3 equivalent.
original type    : Box[Box[x]] forSome x
reduces to       : Box[Box[x]]
type used instead: Box[Box[Any]]
This choice can cause follow-on type errors or hide type errors.
Proceed at own risk.

longer explanation available when compiling with `-explain`
An existential type that came from a Scala-2 classfile cannot be
mapped accurately to to a Scala-3 equivalent.
original type    : Box[Box[x]] forSome x
reduces to       : Box[Box[x]]
type used instead: Box[Box[Any]]
This choice can cause follow-on type errors or hide type errors.
Proceed at own risk.

longer explanation available when compiling with `-explain`
```

## Expectation

Mention which classfile is causing the issue.

**Comment by bishabosha** [comment]
I added some minimised code

**Comment by ckipp01** [comment]
I think this is a duplicate of https://github.com/lampepfl/dotty/issues/13873 and is already fixed by https://github.com/lampepfl/dotty/pull/13879

**Comment by smarter** [comment]
Indeed, fixed in 3.1.2-RC1.
